REPL crash with autocomplete value class member

## Minimized code

```Scala
scala> class TastyFlagSet(val toInt: Int) extends AnyVal { def next: TastyFlagSet = ??? }
// defined class TastyFlagSet

scala> val e = TastyFlagSet(0)
val e: TastyFlagSet = 0

scala> e.n<PRESS-TAB>
```

## Output (click arrow to expand)
<details>

```scala
scala> e.nexception caught when loading module class AnyVal$: java.lang.AssertionError: assertion failed                

Exception in thread "main" java.lang.AssertionError: assertion failed
	at dotty.DottyPredef$.assertFail(DottyPredef.scala:16)
	at dotty.tools.dotc.core.SymDenotations$ClassDenotation.enter(SymDenotations.scala:1756)
	at dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler$.ensureConstructor(Scala2Unpickler.scala:95)
	at dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler$.setClassInfo(Scala2Unpickler.scala:124)
	at dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler$LocalUnpickler.parseToCompletion$1(Scala2Unpickler.scala:582)
	at dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler$LocalUnpickler.complete$$anonfun$1(Scala2Unpickler.scala:608)
	at dotty.runtime.function.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.atReadPos(Scala2Unpickler.scala:325)
	at dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler$LocalUnpickler.complete(Scala2Unpickler.scala:608)
	at dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:260)
	at dotty.tools.dotc.core.Denotations$Denotation.completeInfo$1(Denotations.scala:185)
	at dotty.tools.dotc.core.Denotations$Denotation.info(Denotations.scala:187)
	at dotty.tools.dotc.core.SymDenotations$ClassDenotation.classParents(SymDenotations.scala:1629)
	at dotty.tools.dotc.core.SymDenotations$ClassDenotation.computeBaseData(SymDenotations.scala:1698)
	at dotty.tools.dotc.core.SymDenotations$BaseDataImpl.apply(SymDenotations.scala:2529)
	at dotty.tools.dotc.core.SymDenotations$ClassDenotation.baseData(SymDenotations.scala:1682)
	at dotty.tools.dotc.core.SymDenotations$ClassDenotation.baseClasses(SymDenotations.scala:1689)
	at dotty.tools.dotc.core.SymDenotations$ClassDenotation.traverse$1(SymDenotations.scala:1704)
	at dotty.tools.dotc.core.SymDenotations$ClassDenotation.computeBaseData(SymDenotations.scala:1710)
	at dotty.tools.dotc.core.SymDenotations$BaseDataImpl.apply(SymDenotations.scala:2529)
	at dotty.tools.dotc.core.SymDenotations$ClassDenotation.baseData(SymDenotations.scala:1682)
	at dotty.tools.dotc.core.SymDenotations$ClassDenotation.baseClasses(SymDenotations.scala:1689)
	at dotty.tools.dotc.core.Types$Type.baseClasses(Types.scala:499)
	at dotty.tools.dotc.core.Types$Type.baseClasses(Types.scala:497)
	at dotty.tools.dotc.reporting.diagnostic.messages$NotAMember.candidates$4(messages.scala:324)
	at dotty.tools.dotc.reporting.diagnostic.messages$NotAMember.closest$1(messages.scala:347)
	at dotty.tools.dotc.reporting.diagnostic.messages$NotAMember.<init>(messages.scala:355)
	at dotty.tools.dotc.reporting.diagnostic.messages$NotAMember$.apply(messages.scala:309)
	at dotty.tools.dotc.typer.TypeAssigner.selectionType$$anonfun$4(TypeAssigner.scala:294)
	at dotty.tools.dotc.reporting.diagnostic.MessageContainer.contained(MessageContainer.scala:59)
	at dotty.tools.dotc.reporting.diagnostic.MessageContainer.message(MessageContainer.scala:42)
	at dotty.tools.dotc.reporting.diagnostic.MessageContainer.isNonSensical(MessageContainer.scala:71)
	at dotty.tools.dotc.reporting.HideNonSensicalMessages.isHidden(HideNonSensicalMessages.scala:17)
	at dotty.tools.repl.package$$anon$1.isHidden(package.scala:11)
	at dotty.tools.dotc.reporting.Reporter.report(Reporter.scala:271)
	at dotty.tools.dotc.reporting.Reporting.error(Reporter.scala:136)
	at dotty.tools.dotc.core.Contexts$Context.error(Contexts.scala:75)
	at dotty.tools.dotc.typer.ErrorReporting$.errorType(ErrorReporting.scala:29)
	at dotty.tools.dotc.typer.TypeAssigner.selectionType(TypeAssigner.scala:294)
	at dotty.tools.dotc.typer.Typer.selectionType(Typer.scala:83)
	at dotty.tools.dotc.typer.TypeAssigner.accessibleSelectionType(TypeAssigner.scala:304)
	at dotty.tools.dotc.typer.Typer.accessibleSelectionType(Typer.scala:83)
	at dotty.tools.dotc.typer.TypeAssigner.assignType(TypeAssigner.scala:339)
	at dotty.tools.dotc.typer.Typer.assignType(Typer.scala:83)
	at dotty.tools.dotc.typer.Typer.typedSelect(Typer.scala:495)
	at dotty.tools.dotc.typer.Typer.typeSelectOnTerm$1(Typer.scala:508)
	at dotty.tools.dotc.typer.Typer.typedSelect(Typer.scala:532)
	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2188)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2265)
	at dotty.tools.dotc.typer.Typer.op$1(Typer.scala:2304)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2313)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2316)
	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2360)
	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2382)
	at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:821)
	at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:825)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2216)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2266)
	at dotty.tools.dotc.typer.Typer.op$1(Typer.scala:2304)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2313)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2316)
	at dotty.tools.dotc.typer.Namer.typedAheadExpr$$anonfun$1(Namer.scala:1322)
	at dotty.tools.dotc.typer.Namer.typedAhead(Namer.scala:1312)
	at dotty.tools.dotc.typer.Namer.typedAheadExpr(Namer.scala:1322)
	at dotty.tools.dotc.typer.Namer.rhsType$2(Namer.scala:1456)
	at dotty.tools.dotc.typer.Namer.cookedRhsType$1(Namer.scala:1467)
	at dotty.tools.dotc.typer.Namer.lhsType$1(Namer.scala:1468)
	at dotty.tools.dotc.typer.Namer.inferredType$1(Namer.scala:1479)
	at dotty.tools.dotc.typer.Namer.valOrDefDefSig(Namer.scala:1487)
	at dotty.tools.dotc.typer.Namer$Completer.typeSig(Namer.scala:808)
	at dotty.tools.dotc.typer.Namer$Completer.completeInCreationContext(Namer.scala:929)
	at dotty.tools.dotc.typer.Namer$Completer.complete(Namer.scala:840)
	at dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:260)
	at dotty.tools.dotc.core.Denotations$Denotation.completeInfo$1(Denotations.scala:185)
	at dotty.tools.dotc.core.Denotations$Denotation.info(Denotations.scala:187)
	at dotty.tools.dotc.core.SymDenotations$SymDenotation.ensureCompleted(SymDenotations.scala:400)
	at dotty.tools.dotc.typer.Typer.retrieveSym(Typer.scala:2160)
	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2185)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2265)
	at dotty.tools.dotc.typer.Typer.op$1(Typer.scala:2304)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2313)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2316)
	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2338)
	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2382)
	at dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:1875)
	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2198)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2265)
	at dotty.tools.dotc.typer.Typer.op$1(Typer.scala:2304)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2313)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2316)
	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2338)
	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2382)
	at dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:2001)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2239)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2266)
	at dotty.tools.dotc.typer.Typer.op$1(Typer.scala:2304)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2313)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2316)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2424)
	at dotty.tools.dotc.typer.FrontEnd.liftedTree1$2(FrontEnd.scala:78)
	at dotty.tools.dotc.typer.FrontEnd.typeCheck$$anonfun$1(FrontEnd.scala:83)
	at dotty.runtime.function.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at dotty.tools.dotc.typer.FrontEnd.monitor(FrontEnd.scala:42)
	at dotty.tools.dotc.typer.FrontEnd.typeCheck(FrontEnd.scala:84)
	at dotty.tools.repl.REPLFrontEnd.runOn(ReplFrontEnd.scala:23)
	at dotty.tools.dotc.Run.runPhases$4$$anonfun$4(Run.scala:167)
	at dotty.runtime.function.JProcedure1.apply(JProcedure1.java:15)
	at dotty.runtime.function.JProcedure1.apply(JProcedure1.java:10)
	at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
	at dotty.tools.dotc.Run.runPhases$5(Run.scala:177)
	at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:185)
	at dotty.runtime.function.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:64)
	at dotty.tools.dotc.Run.compileUnits(Run.scala:192)
	at dotty.tools.dotc.Run.compileUnits(Run.scala:139)
	at dotty.tools.repl.ReplCompiler.typeCheck$$anonfun$1(ReplCompiler.scala:284)
	at scala.util.Either.flatMap(Either.scala:352)
	at dotty.tools.repl.ReplCompiler.typeCheck(ReplCompiler.scala:289)
	at dotty.tools.repl.ReplDriver.completions(ReplDriver.scala:177)
	at dotty.tools.repl.ReplDriver.$anonfun$1(ReplDriver.scala:110)
	at org.jline.reader.impl.LineReaderImpl.doComplete(LineReaderImpl.java:3962)
	at org.jline.reader.impl.LineReaderImpl.expandOrComplete(LineReaderImpl.java:3875)
	at org.jline.reader.impl.LineReaderImpl.readLine(LineReaderImpl.java:585)
	at org.jline.reader.impl.LineReaderImpl.readLine(LineReaderImpl.java:404)
	at dotty.tools.repl.JLineTerminal.readLine(JLineTerminal.scala:68)
	at dotty.tools.repl.ReplDriver.readLine$1(ReplDriver.scala:115)
	at dotty.tools.repl.ReplDriver.loop$1(ReplDriver.scala:125)
	at dotty.tools.repl.ReplDriver.runUntilQuit$$anonfun$1(ReplDriver.scala:130)
	at dotty.tools.repl.ReplDriver.withRedirectedOutput(ReplDriver.scala:148)
	at dotty.tools.repl.ReplDriver.runUntilQuit(ReplDriver.scala:130)
	at dotty.tools.repl.Main$.main(Main.scala:6)
	at dotty.tools.repl.Main.main(Main.scala)
```
</details>

--- a/compiler/src/dotty/tools/dotc/ast/DesugarEnums.scala
+++ b/compiler/src/dotty/tools/dotc/ast/DesugarEnums.scala
@@ -94,7 +94,7 @@ object DesugarEnums {
    */
   private def enumScaffolding(implicit ctx: Context): List[Tree] = {
     val valuesDef =
-      DefDef(nme.values, Nil, Nil, TypeTree(), Select(valuesDot(nme.values), nme.toArray))
+      DefDef(nme.values, Nil, Nil, TypeTree(defn.ArrayOf(enumClass.typeRef)), Select(valuesDot(nme.values), nme.toArray))
         .withFlags(Synthetic)
     val privateValuesDef =
       ValDef(nme.DOLLAR_VALUES, TypeTree(),

--- a/compiler/src/dotty/tools/dotc/core/SymDenotations.scala
+++ b/compiler/src/dotty/tools/dotc/core/SymDenotations.scala
@@ -1810,13 +1810,7 @@ object SymDenotations {
      */
     def enter(sym: Symbol, scope: Scope = EmptyScope)(implicit ctx: Context): Unit = {
       val mscope = scope match {
-        case scope: MutableScope =>
-          // if enter gets a scope as an argument,
-          // than this is a scope that will eventually become decls of this symbol.
-          // And this should only happen if this is first time the scope of symbol
-          // is computed, ie symbol yet has no future.
-          assert(this.nextInRun.validFor.code <= this.validFor.code)
-          scope
+        case scope: MutableScope => scope
         case _ => unforcedDecls.openForMutations
       }
       if (proceedWithEnter(sym, mscope)) {

